### PR TITLE
Add CircleCI config for CD of new tags

### LINKED ISSUE #32: Create a process to upload versions of commerce.js to CDN on release
The build process of `commerce.js` creates some bundles that can be used inside traditional `<script>` tags if you don't want to mess around with NPM. We should figure out a process (probably during `npm publish`) that will upload these scripts to a CDN.

### PR DESCRIPTION
**Pull request by ScopeyNZ**
I've done some tests: https://app.circleci.com/github/chec/commerce.js/pipelines

Resolves #32

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,3 +17,35 @@ jobs:
...
+    jobs:
+      - test
+      - deploy:
+          requires:
+            - test
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /.*/
```
Comment (robbieaverill) [inline .circleci/config.yml]: What do these rules mean?

File: .circleci/config.yml [hunk 2]
```diff
@@ -15,3 +17,35 @@ jobs:
...
+      - checkout
+      - run: npm install
+      - run: npm run prepublishOnly
+      - aws-s3/copy:
+          from: lib/commerce.js
+          to: s3://v2/commerce.js
+          arguments: --endpoint $AWS_ENDPOINT
+      - aws-s3/copy:
+          from: lib/commerce.js
+          to: s3://commerce/$CIRCLE_TAG.js
```
Comment (robbieaverill) [inline .circleci/config.yml]: I like this! Nice

File: .circleci/config.yml (hunk 1 above)
Comment (robbieaverill) [reply .circleci/config.yml]: Ignore every branch but build on every tag?

Comment (ScopeyNZ) [reply .circleci/config.yml]: Yes.

File: .circleci/config.yml (hunk 2 above)
Comment (ScopeyNZ) [reply .circleci/config.yml]: Yeah I figured it was super-minimal (storage-wise) but worth doing.

**Comment by ScopeyNZ** [comment]
Note that this will always override the file at `v2/commerce.js` - This should be fine assuming we never tag a prior commit - but maybe I could do some magic to ensure a `1.x` tag doesn't overwrite the `v2` SDK?

**Comment by robbieaverill** [comment]
>but maybe I could do some magic to ensure a 1.x tag doesn't overwrite the v2 SDK?

Yep good call, I didn't think of that

**Comment by robbieaverill** [comment]
Tag regex: `/2\.[.+]/` or something?

**Comment by ScopeyNZ** [comment]
I was thinking why not auto-deploy any major version - but I'm trying to use `sed` for this and it's doing my head in

**Pull request merged**